neutron/convert.cpp

```diff
@@ -257,21 +257,24 @@ T swap_tof_related_labels_and_attrs(T &&x, const Dim from, const Dim to) {
         // a subsequent unit conversion of an item on its own would not be
         // possible. It needs to be determined if there is a better way to
         // handle attributes so this can be avoided.
-        for (const auto &item : iter(x))
-          item.second.attrs().set(field, x.labels()[field]);
+        if constexpr (std::is_same_v<std::decay_t<T>, Dataset>)
```

Comment:
Why is this type check necessary. i.e. What other `T` substitued types are supported?

Reply:
`DataArray`.